nflow-engine/src/main/java/com/nitorcreations/nflow/engine/internal/executor/WorkflowStateProcessor.java

```diff
@@ -185,9 +190,16 @@ private WorkflowInstance saveWorkflowInstanceState(StateExecutionImpl execution,
     WorkflowInstance.Builder builder = new WorkflowInstance.Builder(instance)
       .setNextActivation(execution.getNextActivation())
       .setStatus(getStatus(execution, definition.getState(execution.getNextState())))
-      .setStateText(getStateText(instance, execution))
       .setState(execution.getNextState())
+      .setStateText(getStateText(instance, execution))
       .setRetries(execution.isRetry() ? execution.getRetries() + 1 : 0);
+
+    if(!execution.isStateProcessed()) {
+      workflowInstanceDao.updateWorkflowInstanceAfterExecution(builder.build(), null,
```

Comment:
should we have a update method that takes workflow instance only to avoid passing null & empty list?